hedera-node/src/main/java/com/hedera/services/sigs/order/SigRequirements.java

```diff
@@ -1308,14 +1310,25 @@ private KeyOrderingFailure nftIncludeIfNecessary(
             final List<JKey> required,
             final TokenID token,
             final CryptoTransferTransactionBody op,
-            final @Nullable LinkedRefs linkedRefs) {
+            final @Nullable LinkedRefs linkedRefs,
+            final boolean autoCreationAllowed) {
+        final var isSender = counterparty == null;
         if (!payer.equals(party)) {
             var result = sigMetaLookup.aliasableAccountSigningMetaFor(party, linkedRefs);
             if (!result.succeeded()) {
-                return result.failureIfAny();
+                final var reason = result.failureIfAny();
+                if ((reason == IMMUTABLE_ACCOUNT && !isSender)
```

Comment:
The only `IMMUTABLE_ACCOUNT`'s are `0.0.800` and `0.0.801`, and neither can ever be associated to any token.

So I definitely don't think we should have a special case to return `NONE` here.

The special case on line 1287,
```
    ...
                final var reason = result.failureIfAny();
                final var isCredit = adjust.getAmount() > 0L;
                if (reason == IMMUTABLE_ACCOUNT && isCredit) {
                    return NONE;
    ...
```
is needed because of course `0.0.800` and `0.0.801` can still be credited with hbar.

Reply:
Ah that makes sense ! Will add a comment at the line 1287

Reply:
Added condition for only Hbar transfers `if (reason == IMMUTABLE_ACCOUNT && isCredit && isForHbar) {`